### ISSUE TITLE
CellPose wrapper is too rigid and creates confusing directory structures

### ISSUE THREAD

**Issue by talmo** [talmolab/dreem#170]
## Summary

`run_cellpose_segmentation()` only accepts a directory of individual `.tif` files. To use it with a video (`.mp4`) or TIFF stack, users must manually:

1. Extract frames from the video to individual TIFFs
2. Create the CTC directory structure (`name/`, `name_GT/TRA/`)
3. Run CellPose on the extracted frames
4. Track the deeply nested output paths through to the tracking step

The resulting directory structure is confusing and deeply nested:

```
project/
├── well7f5merge_Crop/
│   ├── well7f5merge_Crop/           ← raw frames (repeated name!)
│   │   ├── frame_00000.tif
│   │   └── ...
│   └── well7f5merge_Crop_GT/
│       └── TRA/                     ← masks
│           ├── frame_00000.tif
│           └── ...
├── results/                         ← tracking output (separate location)
│   └── well7f5merge_Crop.dreem_inference.*.tif
└── script.py
```

## Current limitations

1. **Input format**: `run_cellpose_segmentation()` only accepts a directory of `.tif`/`.tiff` files. CellPose's `model.eval()` accepts numpy arrays directly — the file I/O restriction is unnecessary.

2. **Manual frame extraction**: Users must convert videos/TIFF stacks to individual frames themselves before calling the function. The notebook (`examples/microscopy-demo.ipynb`) includes ~15 lines of boilerplate for this.

3. **Rigid CTC directory structure**: `get_ctc_paths()` in `config.py` hardcodes the `_GT/TRA/` naming convention. The raw frames directory and mask directory must follow exact naming (`name/` and `name_GT/TRA/`).

4. **No output path management**: Users must manually construct and track all intermediate paths.

### 1. Flexible `run_cellpose_segmentation()` input handling

```python
def run_cellpose_segmentation(
    data_path: str | np.ndarray,  # dir, .tif stack, .mp4/.avi, numpy array, or sio.Video
    output_path: str | None = None,  # if None, auto-derive from data_path
    diameter: int = 25,
    gpu: bool = True,
    cellprob_threshold: float = 0.0,
) -> np.ndarray:
```

Auto-detect input format and normalize to a frame list internally:
- **Directory of TIFFs** → load as stack (current behavior)
- **TIFF stack** (single `.tif` file) → `tifffile.imread()`
- **Video** (`.mp4`, `.avi`) → `sio.load_video()`
- **Numpy array** → use directly
- **`sio.Video` object** → iterate frames

Handle grayscale conversion internally (`frame[..., 0]` for RGB) instead of requiring the user to do it.

### 2. Simplify CTC directory management

Options (not mutually exclusive):

**a) Add a helper function:**
```python
def setup_ctc_dirs(data_path: str, output_dir: str | None = None) -> dict:
    """Create CTC directory structure and return paths.
    
    Returns:
        {"frames_dir": ..., "masks_dir": ..., "dataset_dir": ..., "results_dir": ...}
    """
```

**b) Relax `get_ctc_paths()` to accept explicit paths:**
Instead of requiring the `_GT/TRA/` naming convention, allow users to pass frame and mask directories explicitly.

**c) Accept arrays directly in the inference pipeline:**
Skip the CTC directory structure entirely for simple use cases — accept `(raw_frames, mask_frames)` arrays directly in the tracking pipeline.

### 3. Ideal end-to-end API

The goal should be something like:
```python
from dreem.utils import run_cellpose_segmentation
from dreem.inference.track import run as run_tracking

# One function call for segmentation — accepts any format
masks = run_cellpose_segmentation("my_video.mp4", diameter=25)

# One function call for tracking — no directory structure required
result = run_tracking(checkpoint="microscopy", frames="my_video.mp4", masks=masks, crop_size=25)
```

## Files involved

- `dreem/utils/run_cellpose_segmentation.py` — the CellPose wrapper (main target)
- `dreem/io/config.py:get_ctc_paths()` (lines 188-224) — rigid CTC directory discovery
- `dreem/datasets/cell_tracking_dataset.py` — CTC dataset loader
- `examples/microscopy-demo.ipynb` — contains frame extraction boilerplate that should be in the library
